### PR TITLE
feat(builder): validate operators at startup time

### LINKED ISSUE #73: Throw at startup time instead of run time in case of wrong parameters
**What is the issue about**:
early throw in case of wrong parameters

**What happened**:
if I defined a handler function of type: 
```
func(ctx context.Context, first int, after string)
```

while using definitions as 
```
Parameters: []definition.Parameter{
					{
						Source: definition.Query,
						Name:   "after",
						Type:   reflect.TypeOf(""),
					},
					{
						Source:  definition.Query,
						Name:    "first",
						Type:    reflect.TypeOf(0),
						Default: 10,
					},
				},
```

then there's only an error thrown during request time.

But in fact, this check can be done at startup time.

**What you expected to happen**:

**Anything else we need to know**:

### PR DESCRIPTION
**Pull request by kdada**
**What this PR does / why we need it**:
For an operator chain:
```
in -> operators[0].In()
operators[0].Out() -> operators[1].In()
...
operators[N].Out() -> out
```
Parameter types and result types are checked at startup time now.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #73 

**Special notes for your reviewer**:
/assign @yejiayu 
/assign @Jimexist 
/area framwork
/kind enhancement
/priority P1

**Release note**:

```release-note
Check parameters and results of operators at startup time
```

### CODE REVIEW

File: service/executor_test.go [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+			definition.Definition{
+				Method: definition.Method(""),
+			},
+			definitionNoMethod,
+		},
+		{
+			definition.Definition{
+				Method: definition.Get,
+			},
+			definitionNoComsumes,
```
Comment (ddysher) [inline service/executor_test.go]: typo: definitionNoConsumes

**Comment by kdada** [comment]
@ddysher PTAL

File: service/executor.go [hunk 2]
```diff
@@ -202,14 +208,23 @@ func (i *inspector) generateResults(funcName string, typ reflect.Type, rs []defi
...
 		}
 		result := result{
 			index:     index,
 			handler:   handler,
 			operators: r.Operators,
 		}
-		if err := handler.Validate(typ.Out(index)); err != nil {
+		outType := typ.Out(index)
+		if len(result.operators) > 0 {
+			LastOperatorOutType := result.operators[len(result.operators)-1].Out()
```
Comment (yejiayu) [inline service/executor.go]: What if `result.operators` > 1?

**Pull request merged**